src/sync/outgoingSyncUtils.js

```diff
@@ -51,16 +52,37 @@ export function generateSyncJson(database, settings, syncOutRecord) {
   } else {
     // Get the record the syncOutRecord refers to from the database
     const recordResults = database.objects(recordType).filtered('id == $0', recordId);
-    if (!recordResults || recordResults.length === 0) { // No such record
+    if (!recordResults || recordResults.length === 0) {
+      // No such record
       throw new Error(`${recordType} with id = ${recordId} missing`);
-    } else if (recordResults.length > 1) { // Duplicate records
+    } else if (recordResults.length > 1) {
+      // Duplicate records
       throw new Error(`Multiple ${recordType} records with id = ${recordId}`);
     }
     const record = recordResults[0];
 
     // Generate the appropriate data for the sync object to carry, representing the
     // record in its upstream form
-    syncData = generateSyncData(settings, recordType, record);
+    try {
+      syncData = generateSyncData(settings, recordType, record);
+    } catch (error) {
+      // There was an error with data, often a null object
+      let message;
+      const siteName = settings.get(SYNC_SITE_NAME);
+      const syncUrl = settings.get(SYNC_URL);
+      const originalMessage = error.message;
+      // Change error message to be helpful in bugsnag
+      message =
+        `SYNC OUT ERROR. siteName: ${siteName}, serverUrl: ${syncUrl}, ` +
+        `syncOutRecord.id: ${syncOutRecord.id}, storeId: ${storeId} changeType: ${changeType}, ` +
+        `recordType: ${recordType}, recordId: ${recordId}, message: ${originalMessage}`;
+      // Ping the error off to bugsnag
+      bugsnagClient.notify(Object.assign({}, error, message));
+
+      // Make a nicer message for users and throw it again.
+      message = `There was an error syncing. Contact mSupply mobile support. ${originalMessage}`;
+      throw Object.assign({}, error, message); // doesn't like object literal syntax
```

Comment:
I know i've used Object.assign() quite a bit in other projects, but have been changing it to {...one, ...two} every time I see it now.  

Also I think you are throwing a json here not an Error, and passing json to bugsnag.notify, i think it should still work fine, except no stack trace

Reply:
1. Eslint was whining about throw + object literal syntax. My comment could explain that better. The object destructuring was my first instinct too. Might revert back and just ignore line it (unless the eslint rules expla

2. Yea I need to test this. Errors are objects, I was hoping that making a copy would include the propertyholding the stack trace but that's a presumption. Hence needing testing! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/EvalError

Reply:
Object assign or destructuring won't work on Error object.
<img width="640" alt="screen shot 2018-02-14 at 11 37 54 am" src="https://user-images.githubusercontent.com/26194949/36177873-f4006020-117b-11e8-92d8-2eb44ac73439.png">

Reply:
Alright I've been doing testing. Clone an Error object in anyway and it loses it's special sauce and bugsnag throws a wobbly about it.

Looking at the their implementation [(yay for opensource)](https://github.com/bugsnag/bugsnag-react-native/blob/master/src/Bugsnag.js#L311-L321) it appears it makes it's own internal report object rather than keeping a reference the error.

I'm just going to revert to editing the error message... or use their notify API properly.